components/net_traits/response.rs

```diff
@@ -135,11 +135,18 @@ impl Response {
 
     pub fn is_network_error(&self) -> bool {
         match self.response_type {
-            ResponseType::Error => true,
+            ResponseType::Error(..) => true,
             _ => false
         }
     }
 
+    pub fn get_network_error(&self) -> Option<&NetworkError> {
```

Comment:
I think it's better to at least make it possible for new callers to not have to clone, if they don't need to.